packages/format-library/src/link/index.js

```diff
@@ -38,10 +38,38 @@ function Edit( {
 	onFocus,
 	contentRef,
 } ) {
-	const [ addingLink, setAddingLink ] = useState( false );
+	const [ editingLink, setEditingLink ] = useState( false );
+	const [ creatingLink, setCreatingLink ] = useState( false );
+
 	// We only need to store the button element that opened the popover. We can ignore the other states, as they will be handled by the onFocus prop to return to the rich text field.
 	const [ openedBy, setOpenedBy ] = useState( null );
 
+	const [ autoFocus, setAutoFocus ] = useState( true );
+
+	function setIsEditingLink( _val, { shouldAutoFocus = true } = {} ) {
+		setEditingLink( _val );
+		setAutoFocus( shouldAutoFocus );
```

Comment:
why not set the state right where we determine the value of `shouldAutoFocus`?

Reply:
Given how critical autoFocus state is to the UX and a11y I wanted to make it as hard as possible for someone to accidentally call `setEditingLink` without managing the focus. 

So I made this wrapper "setter" function to default autoFocus to `true` meaning that someone has to really really want to disable autoFocus.

It's not perfect but...